x-pack/plugins/serverless_search/public/plugin.ts

```diff
@@ -33,6 +34,25 @@ export class ServerlessSearchPlugin
       ServerlessSearchPluginStartDependencies
     >
 {
+  private async findAndCreateDataView(data: DataPublicPluginStart) {
+    const dataViewExists =
+      (await data.dataViews.find('default:all-data'))
```

Comment:
Would it make more sense to use a specific id for this data view?

Reply:
Currently dataView creates generated id which makes it difficult to filter out data views based on Id. 

 So came up with creating dataview with custom id  and get dataview based on that id. 

For example:

```
const dataViewExists = await dataViews.get('default_all_data_id').catch(() => false);
if (!dataViewExists) {
   const defaultDataViewExists = await dataViews.defaultDataViewExists();
   if (!defaultDataViewExists) {
     await dataViews.createAndSave({
        allowNoIndex: false,
        name: 'default:all-data',
        title: '*',
        id: 'default_all_data_id',
      });
 }
}
```

Although, I do see an error in the console. Which makes me wonder if there would be any side effects of creating a Dataview with id which is not a Saved Object id. 

<img width="610" alt="Screenshot 2023-11-02 at 11 32 14 AM" src="https://github.com/elastic/kibana/assets/55930906/d6494c28-a591-4fb7-b468-c3c1638ef238">

Reply:
I think that's just verification that the data view id isn't already in use.

Yes, supplying an id such as you have here is the best way to do this.

Reply:
Added in 801bd5d3aceea4e9ed78d8c984afaa7a06ff52e9